packages/demo/src/configuration/blocks/effects/lutBlock.ts

```diff
@@ -0,0 +1,176 @@
+import type { Effect } from "@babylonjs/core/Materials/effect";
+import type { SmartFilter, IDisableableBlock, RuntimeData } from "@babylonjs/smart-filters";
+import {
+    ShaderBlock,
+    ConnectionPointType,
+    ShaderBinding,
+    injectDisableUniform,
+    createStrongRef,
+} from "@babylonjs/smart-filters";
+import { BlockNames } from "../blockNames";
+import { Vector2 } from "@babylonjs/core";
+
+const shaderProgram = injectDisableUniform({
+    fragment: {
+        uniform: `
+            uniform sampler2D _input_;
+            uniform sampler2D _lutTexture_;
+            uniform float _level_;
+            uniform vec2 _blueSlicesIn2DLayout_;
+            uniform vec2 _texSize_;
+            `,
+
+        mainFunctionName: "_lutFilter_",
+
+        mainInputTexture: "_input_",
+
+        functions: [
+            {
+                name: "_lutFilter_",
+                code: `
+                vec4 _lutFilter_(vec2 vUV) {
+                    vec3 color = texture2D(_input_, vUV).rgb;
+                    vec4 tempColor = _getColorInLookupTable_(color, _lutTexture_, _blueSlicesIn2DLayout_, _texSize_);
```

Comment:
what does blueSlicesIn2DLayout refer to

Reply:
It refers to the row/column of blue squares in a lut image. I've hardcoded it here since Havana filters have all the same config.